ui/next.config.js

```diff
@@ -4,13 +4,18 @@ const nextConfig = {
 
     async redirects() {
         return [
-          {
-            source: '/search',
-            destination: '/search/publications/',
-            permanent: true,
-          },
-        ]
-      },
+            {
+                source: '/search',
+                destination: '/search/publications/',
+                permanent: true
+            },
+            {
+                source: '/publications/:path',
```

Comment:
This type of redirect is ok for server-side redirects, eg: when the user tries to access a publication URL directly.
But, it's not working properly with client-side redirects using next/router and next/link. All the links and other client-side redirects like `router.push(...)` pointing to `/publications/${id}` will trigger a hard page reload because of this redirect. While redirecting, the page is blank for a few seconds and it's loading very very slow. This a known problem with Next.js when using redirects, which unfortunately won't be fixed any time soon. I've came into this problem when I added the redirect from /publications to /publications/id/versions/latest.

I see two potential fixes for this.

1. To remove this redirect and leave the publication page + redirect from getServerSideProps as it was because redirecting from getServerSideProps works fine, without hard reloads (which is also interesting, but... it's Next.js...). The publication page can be blank, it doesn't need to fetch or render anything, it just needs to return a redirect from getServerSideProps.

2. to keep this redirect for server-side redirects and update all links and redirects which currently point to /publications/{id} to point directly to the latest version of that publication /publication/{id}/versions/latest

Reply:
Thanks for the explanation - I thought fix 1 would be the best so I've moved it over to getServerSideProps in the latest commit.